src/Renderer/PointsMaterial.js

```diff
@@ -26,19 +35,24 @@ class PointsMaterial extends RawShaderMaterial {
         if (__DEBUG__) {
             this.defines.DEBUG = 1;
         }
-
-        this.updateUniforms();
     }
 
     enablePicking(pickingMode) {
         // we don't want pixels to blend over already drawn pixels
         this.blending = pickingMode ? NoBlending : NormalBlending;
-        this.uniforms.pickingMode.value = pickingMode;
+        if (pickingMode) {
+            this.oldMode = this.uniforms.mode.value;
```

Comment:
Shouldn't `this.oldMode` be also assigned in the constructor (like @elemoine mentionned here https://github.com/iTowns/itowns/pull/719#issuecomment-380376078)?

Reply:
Yes, at least initialized to an empty value, if we want to rely entirely on the update mechanism.